### PR TITLE
created loading todos logic for Todo app

### PR DESCRIPTION
**Pull request by OPTIMISTIXX**
[DEMO LINK](https://OPTIMISTIXX.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+  const [isErrorsClosed, setIsErrorsClosed] = useState(true);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const result = await getTodos(USER_ID);
+
+        if (!Array.isArray(result) || !result.length) {
+          throw new Error('incorrect url');
+        }
```
[the remark this answers is not in the document]
Comment (anastasiiavorobiova) [reply src/App.tsx:26]: The error object looks different here. You can check status code to check the response status before setting todos

[the remark this answers is not in the document]
Comment (anastasiiavorobiova) [reply src/App.tsx:26]: Screenshot 2024-01-03 at 16 01 12

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

File: src/App.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (OPTIMISTIXX) [reply src/App.tsx]: > The error object looks different here. You can check status code to check the response status before setting todos
response is an array of todos. He hasnt a status property. Please, show me, how can I check my code by status here?

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    >
+      <span className="todo-count" data-cy="TodosCounter">
+        3 items left
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={`filter__link ${tasksFilter === 'all' && 'selected'}`}
```
Comment (volodymyr-soltys97) [inline src/components/Footer.tsx:38]: Use a classnames library for add classes with condition

File: src/components/Section.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  switch (tasksFilter) {
+    case TasksFilter.active:
+      filteringTodos = todos.filter((todo) => !todo.completed);
+      break;
+    case TasksFilter.completed:
+      filteringTodos = todos.filter((todo) => todo.completed);
+      break;
+    default:
+      filteringTodos = todos;
```
Comment (volodymyr-soltys97) [inline src/components/Section.tsx:23]:
```suggestion
    case TasksFilter.active:
      filteringTodos = todos.filter((todo) => !todo.completed);
      break;
      
    case TasksFilter.completed:
      filteringTodos = todos.filter((todo) => todo.completed);
      break;
      
    default:
      filteringTodos = todos;
```

File: src/components/Section.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              type="checkbox"
+              className="todo__status"
+            />
+          </label>
+
+          <span
+            data-cy="TodoTitle"
+            className="todo__title"
+          >
+            {todo.title}
```
Comment (volodymyr-soltys97) [inline src/components/Section.tsx:46]: Use a destruction for `todo`

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

**Review by IvanFesenko — APPROVED** [review]
Well done.

Note: Button status is incorrect
2024-01-04_16h46_00

**Review by IvanFesenko — APPROVED** [review]

**Comment by OPTIMISTIXX** [comment]
> Well done.
> 
> Note: Button status is incorrect 2024-01-04_16h46_00

**Pull request closed without merging**